### PR TITLE
[TM-4919] Bidding Tool Integration

### PR DESCRIPTION
**Pull request by jlam-mpc**
Updates:
- 

Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/881)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/420)

[TM-4919](https://metaphase.atlassian.net/browse/TM-4919)

[TM-4919]: https://metaphase.atlassian.net/browse/TM-4919?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx [hunk 1]
```diff
@@ -7,22 +7,23 @@ import TextareaAutosize from 'react-textarea-autosize';
 import swal from '@sweetalert/with-react';
 import PropTypes from 'prop-types';
 import shortid from 'shortid';
-import { getResult, userHasPermissions } from 'utilities';
+import { isEqual } from 'lodash';
```
Comment (M00staff) [inline src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:10]: 🌟

IMO - this is the ONLY Lodash function worth using.

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx [hunk 2]
```diff
@@ -7,22 +7,23 @@ import TextareaAutosize from 'react-textarea-autosize';
...
 import { Row } from 'Components/Layout';
 import { Definition } from '../../../DefinitionList';
-import { biddingTool, biddingToolCreate, biddingToolEdit } from '../../../../actions/biddingTool';
+import { biddingTool, biddingToolCreate, biddingToolCreateData, biddingToolDelete, biddingToolEdit } from '../../../../actions/biddingTool';
 import Spinner from '../../../Spinner/Spinner';
 import CheckBox from '../../../CheckBox/CheckBox';
 import { history } from '../../../../store';
 
-
+// eslint-disable-next-line
```
Comment (M00staff) [inline src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:20]: we should get rid of this lint rule ... there are a few others i dislike - i might make a list 🤔

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx [hunk 3]
```diff
@@ -32,58 +33,142 @@ const BiddingToolCard = (props) => {
...
+
+    rr_point: result?.rr_point ?? '',
+    medical: result?.medical ?? '',
+    remarks: result?.remarks ?? '',
+    quarters_remark: result?.quarters_remark ?? '',
+    special_ship_allowance: result?.special_ship_allowance ?? '',
+    school_year: result?.school_year ?? '',
+    grade_education: result?.grade_education ?? '',
+    efm_employment: result?.efm_employment ?? '',
+  };
```
Comment (M00staff) [inline src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:124]: 👍

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx [hunk 4]
```diff
@@ -149,157 +236,76 @@ const BiddingToolCard = (props) => {
...
-    }));
   };
 
   const onCancel = () => {
     if (isCreate) history.push(rootLocation());
     else if (setEditMode) setEditMode(false);
-    swal.close();
   };
   const showCancelModal = () => {
+    if (isEqual(initialValues, values)) {
```
Comment (M00staff) [inline src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:259]: 👍

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx [hunk 5]
```diff
@@ -149,157 +236,76 @@ const BiddingToolCard = (props) => {
...
+          <div className="simple-action-modal">
+            <div className="help-text">
+              {isCreate ?
+                <span>Are you sure you want to discard this Bidding Tool draft?</span> :
+                <span>Are you sure you want to discard all changes made to this Bidding Tool?</span>
+              }
+            </div>
+            <div className="modal-controls">
+              <button onClick={() => { onCancel(); swal.close(); }}>Yes</button>
+              <button className="usa-button-secondary" onClick={() => swal.close()}>No</button>
```
Comment (M00staff) [inline src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:276]: thank you for updating the wording on this modal - MUCH more clear!

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx [hunk 6]
```diff
@@ -313,103 +319,115 @@ const BiddingToolCard = (props) => {
...
               id="tod"
-              defaultValue={tod}
-              onChange={(e) => setTod(e.target.value)}
+              defaultValue={values.tod}
+              onChange={(e) => setValues({ ...values, tod: e.target.value })}
             >
-              {dummySelectOptions.map(b => (
-                <option value={b.code}>{b.name}</option>
+              {tods?.map(b => (
+                <option key={b.code} value={b.code}>{b.description}</option>
```
Comment (M00staff) [inline src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:368]: Should this dropdown also handle OTHER tod's - the way you can when editing an Agenda Item?

Will reach out on slack about this

**Comment by M00staff** [comment]
when i log in as Leah, the CDO and click Bidding Tool - page looks like this

... is that expected?

image

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx (hunk 6 above)
Comment (M00staff) [reply src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:368]: i think you might be fine here, actually. 

looking at the spreadsheet and it looks like you only pass in the code and no other value

File: src/actions/biddingTool.js [hunk 7]
```diff
@@ -96,13 +39,33 @@ export function biddingToolFetchDataSuccess(results) {
...
+          dispatch(biddingToolFetchDataErrored(false));
+          dispatch(biddingToolFetchDataLoading(false));
+        });
+      })
+      .catch(() => {
+        batch(() => {
+          dispatch(biddingToolFetchDataErrored(true));
+          dispatch(biddingToolFetchDataLoading(false));
+        });
+      });
```
Comment (M00staff) [inline src/actions/biddingTool.js:68]: was this the try/catch you were having an issue with, the other day?

if so - what was the problem/solution?

**Review by M00staff — APPROVED** [review]
just a few questions - this is great stuff, kudos!

File: src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx (hunk 6 above)
Comment (jlam-mpc) [reply src/Components/BiddingFunctionsPage/BiddingTool/BiddingToolCard/BiddingToolCard.jsx:368]: Based on the mockups and payload of options I do believe this is all I could do with TOD.

**Pull request merged**